heutelbeck/sapl-policy-engine
# Eclipse plugin

## Description

This PR adds an eclipse plugin for the sapl policy language. It uses the support from eclipse support from xtext (https://eclipse.dev/Xtext/documentation/310_eclipse_support.html).
The plugin provides features like syntax highlighting, content assist, an outline, folding and shows errors in the policy file.

The PR adds a project which generates an eclipse update site in eclipse-plugin/sapl-eclipse-repository/target/repository which can be manually added to eclipse. The plugin can be installed from this update site. I tested the installation and usage of the plugin under linux and windows.

Out of scope:
- This PR only includes a plugin for sapl policy files not for sapl-test. Support for sapl-test can be added in a follow-up PR.

How to test:

- Build the project with `mvn clean install` locally.
- Open the "Install" dialog in eclipse (Help -> Install New Software...).
- Add the local update-site  to eclipse (Click on "Add..." , then on "Local..." and select the local update-site. Its in the local git repository in the subfolder "eclipse-plugin/sapl-eclipse-repository/target/repository")
- Select the update-site in the dropdown after "Work with". Its name is "SAPL Eclipse update site".
- Select "SAPL in the table below.
- Click on "Next".
- Click on "Finish".
- Trust all bundles for the sapl plugin in the "Trust Artifacts" dialog. ("Select All" and then "Trust Selected")
- Restart eclipse from the "Software Updates" dialog. ("Restart Now")
- Open a sapl file.

Known problems which can be fixed later and should not block this PR:
- auto formatting isn't using the preferred formatting style yet.
- outline view contains some unnamed nodes.

## Checklist



* [X] I ran all checks which produced no new errors or warnings for my changes.
